ui/src/components/SecretForm/index.jsx

```diff
@@ -130,8 +130,12 @@ export default class SecretForm extends Component {
   };
 
   render() {
-    const { classes, isNewSecret, loading } = this.props;
+    const { secret, classes, isNewSecret, loading } = this.props;
     const { secretName, editorValue, expires, showSecret } = this.state;
+    const isSecretDirty =
+      secretName !== secret.name ||
```

Comment:
This would probably not work when creating a new secret (http://localhost:5080/secrets/create). To fix this, we would need to add a check for `isNewSecret`:

```jsx
const isSecretDirty =
  isNewSecret ||
 // same as before
```